pages/api/transactions/[id].js

```diff
@@ -1,13 +1,24 @@
 import dbConnect from "@/db/connect.js";
 import Transaction from "@/db/models/Transaction";
+import { getServerSession } from "next-auth/next";
+import { authOptions } from "../auth/[...nextauth]";
 
 export default async function handler(request, response) {
-	await dbConnect();
-	const { id } = request.query;
+  const session = await getServerSession(request, response, authOptions);
+  if (!session) {
+    return response.status(401).json({ status: "Not authorized" });
+  }
 
-	try {
-		if (request.method === "GET") {
-			const transaction = await Transaction.findByID(id);
+  await dbConnect();
+  const { id } = request.query;
+
+  if (!id) {
+    return;
+  }
```

Comment:
in theory this should always be true, since the route is only accessible if id has a value.